src/app/public/modules/overlay/overlay.service.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  ComponentRef,
+  Injectable
+} from '@angular/core';
+
+import {
+  NavigationStart,
+  Router
+} from '@angular/router';
+
+import {
+  Subscription
+} from 'rxjs';
+
+import {
+  SkyDynamicComponentService
+} from '../dynamic-component';
+
+import {
+  SkyOverlayAdapterService
+} from './overlay-adapter.service';
+
+import {
+  SkyOverlayConfig
+} from './overlay-config';
+
+import {
+  SkyOverlayHostComponent
+} from './overlay-host.component';
+
+import {
+  SkyOverlayInstance
+} from './overlay-instance';
+
+/**
+ * This service is used to create new overlays.
+ */
+@Injectable()
+export class SkyOverlayService {
+
+  private host: ComponentRef<SkyOverlayHostComponent>;
+
+  private overlays: SkyOverlayInstance[] = [];
+
+  private routerSubscription: Subscription;
+
+  constructor(
+    private dynamicComponentService: SkyDynamicComponentService,
+    private router: Router,
+    private adapter: SkyOverlayAdapterService
+  ) {
+    this.createHostComponent();
+  }
+
+  /**
+   * Creates a new overlay.
```

Comment:
Since this is eventually going to populate our docs, should we make it clear that this only creates shell of the overlay? To populate it, consumers will have to leverage`SkyOverlayInstace`'s `attachComponent()` or `attachTemplate()` methods.

Reply:
Good call.